superglue/lib/index.tsx

```diff
@@ -5,15 +5,22 @@ import { config } from './config'
 import { urlToPageKey, ujsHandlers, argsForHistory } from './utils'
 import { saveAndProcessPage } from './action_creators'
 import { HISTORY_CHANGE, SET_CSRF_TOKEN } from './actions'
+import { ConnectedComponent, Provider, connect } from 'react-redux'
 import {
   combineReducers,
-  createStore,
+  legacy_createStore,
```

Comment:
In this PR I also upgraded redux as `devDepedency` and it seems that `createStore` is deprecated. The recommendation to superglue users is to override the method `buildStore` so they can provide their own. The generators at `https://github.com/thoughtbot/superglue/blob/main/superglue_rails/lib/install/templates/web/application.js` will do that already.

The `buildStore` method in ApplicationBase, the one that I recommend to override, is the only reason Superglue has a dependency on Redux. I could remove that method, and `throw new NotImplementedError` just like the `visitAndRemote` below. I think it wouldn't break any apps since its being overridden anyway.

Reply:
This also makes the case for the abstract class! Thanks so much @pangolingo !